src/fides/api/models/privacy_experience.py

```diff
@@ -363,81 +580,19 @@ def get_privacy_notices_by_region_and_component(
     """
     return (
         db.query(PrivacyNotice)
-        .filter(PrivacyNotice.regions.overlap(regions))  # type: ignore[attr-defined]
+        .join(ExperienceNotices, PrivacyNotice.id == ExperienceNotices.notice_id)
+        .join(
+            PrivacyExperience,
+            PrivacyExperience.experience_config_id
+            == ExperienceNotices.experience_config_id,
+        )
         .filter(
-            or_(
-                and_(
-                    component == ComponentType.overlay,
-                    PrivacyNotice.displayed_in_overlay,
-                ),
-                and_(
-                    component == ComponentType.privacy_center,
-                    PrivacyNotice.displayed_in_privacy_center,
-                ),
-            )
+            PrivacyExperience.region.in_(regions),
+            PrivacyExperience.component == component,
         )
```

Comment:
*Where* a notice is shown is now defined on the experience side of things instead of the notice side of things, so the query here has changed.